starter-code/src/memory.js

```diff
@@ -1,14 +1,53 @@
-// var MemoryGame = function (cards) {
-//   this.cards = cards;
-// };
+var MemoryGame = function (cards) {
```

Comment:
I tried your game and it seems like a lot of bugs where present.
Here is the solution so you can compare the differences:

## Main.js 
```js
var cards = [
  { name: 'aquaman', img: 'aquaman.jpg' },
  { name: 'batman', img: 'batman.jpg' },
  { name: 'captain america', img: 'captain-america.jpg' },
  { name: 'fantastic four', img: 'fantastic-four.jpg' },
  { name: 'flash', img: 'flash.jpg' },
  { name: 'green arrow', img: 'green-arrow.jpg' },
  { name: 'green lantern', img: 'green-lantern.jpg' },
  { name: 'ironman', img: 'ironman.jpg' },
  { name: 'spiderman', img: 'spiderman.jpg' },
  { name: 'superman', img: 'superman.jpg' },
  { name: 'the avengers', img: 'the-avengers.jpg' },
  { name: 'thor', img: 'thor.jpg' },
  { name: 'aquaman', img: 'aquaman.jpg' },
  { name: 'batman', img: 'batman.jpg' },
  { name: 'captain america', img: 'captain-america.jpg' },
  { name: 'fantastic four', img: 'fantastic-four.jpg' },
  { name: 'flash', img: 'flash.jpg' },
  { name: 'green arrow', img: 'green-arrow.jpg' },
  { name: 'green lantern', img: 'green-lantern.jpg' },
  { name: 'ironman', img: 'ironman.jpg' },
  { name: 'spiderman', img: 'spiderman.jpg' },
  { name: 'superman', img: 'superman.jpg' },
  { name: 'the avengers', img: 'the-avengers.jpg' },
  { name: 'thor', img: 'thor.jpg' },
];

$(document).ready(function() {
  var memoryGame = new MemoryGame(cards);
  memoryGame.shuffleCard(memoryGame.cards);
  var html = '';
  memoryGame.cards.forEach(function(pic, index) {
    html += '<div class= "card" id="card_' + pic.name + '">';
    html += '<div class="back"';
    html += '    name="' + pic.img + '">';
    html += '</div>';
    html += '<div class="front" ';
    html += 'style="background: url(img/' + pic.img + ') no-repeat">';
    html += '</div>';
    html += '</div>';
  });

  // Add all the div's to the HTML
  document.getElementById('memory_board').innerHTML = html;
  // Bind the click event of each element to a function
  $('.back').on('click', function() {
    var nameOfImage = $(this).attr('name');
    $(this).addClass('just-clicked');
    $(this).addClass('blocked');
    memoryGame.currentPair.push(nameOfImage);
    $(this).css('background', 'url(img/' + nameOfImage + ')');

    if (memoryGame.currentPair.length === 2) {
      memoryGame.checkIfPair(
        memoryGame.currentPair[0],
        memoryGame.currentPair[1]
      );
    }
  });
});
```
## Memory.js
```js
var MemoryGame = function(cards) {
  this.cards = cards;
  this.currentPair = [];
  this.pairsClicked = 0;
  this.pairsGuessed = 0;
};

MemoryGame.prototype.shuffleCard = function(cardsArr) {
  // var shuffledArray = [];
  // cardsArr.forEach(function(theCard, theIndex) {
  //   var randomCardIndex = Math.floor(Math.random() * cardsArr.length);
  //   shuffledArray.push(cardsArr[randomCardIndex]);
  //   cardsArr.splice(randomCardIndex, 1);
  // });
  // this.cards = shuffledArray;

  this.cards = _.shuffle(cardsArr);
};

MemoryGame.prototype.checkIfPair = function(firstCard, secondCard) {
  this.pairsClicked++;
  $('#pairs_clicked').text(this.pairsClicked);

  if (firstCard === secondCard) {
    $('.back').addClass('blocked');
    $('.just-clicked').addClass('perma-blocked');
    this.pairsGuessed++;
    $('#pairs_guessed').text(this.pairsGuessed);
    $('.just-clicked').removeClass('just-clicked');
    $('.back').removeClass('blocked');
  } else {
    $('.back').addClass('blocked');

    setTimeout(function() {
      $('.just-clicked').css('background', '#456783');
      $('.just-clicked').removeClass('just-clicked');
      $('.back').removeClass('blocked');
    }, 500);
  }
  this.currentPair = [];
  this.finished();
};

MemoryGame.prototype.finished = function() {
  if (this.pairsGuessed > 11) {
    alert('YOU WON!!!!');
  }
};
```